SU-SWS/ansible-playbooks
# SITES-928

# READY FOR REVIEW

# Summary
- Split out the `sites.json` work into its own role
- Remove the `check-php` role from playbooks, as we rarely use it and it is slow

# Needed By (Date)
- 9.28.18

# Criticality
- How critical is this PR on a 1-10 scale? 9/10
- It will be critical for creating sites quickly enough on ACSF, and migrating them quickly

# Steps to Test

1. Check out this branch
2. Run `full-migration-playbook.yml` with [this inventory](https://github.com/SU-SWS/ansible-playbooks/files/2421302/PR58.txt)
3. Ensure that the vhosts are added as custom domains
4. Ensure that the caches are cleared
5. Ensure that the `check-php` role is skipped
6. Run `migration-sync-only-playbook.yml`
7. Ensure that the vhosts are added as custom domains
8. Ensure that the caches are cleared
9. Delete one or more of the sites
10. Run `acsf-site-setup-playbook.yml`
11. Ensure that the deleted sites are set up successfully

# Affected Projects or Products
- ACSF

# Associated Issues and/or People
- JIRA tickets:
    - https://stanfordits.atlassian.net/browse/SITES-928
    - https://stanfordits.atlassian.net/browse/SITES-925

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
